src/main/java/ui/UI.java

```diff
@@ -248,6 +258,8 @@ public void quit() {
             Platform.exit();
             System.exit(0);
         }
+        provider.reset();
+        provider.stop();
```

Comment:
What happens if this part doesn't execute? Does the hotkey process or something persist? Reason is that if `System.exit` is called, this is unreachable. Might be better to put it at the top of the method.

Also we should put this in its own method, so it's clear what it does.